src/Compilers/CSharp/Test/Semantic/Semantics/Utf8StringsLiteralsTests.cs

```diff
@@ -1739,6 +1832,150 @@ static void Main()
+        {
+            var source = @"
+using System;
+class C
+{
+    static void Main()
+    {
+        System.Console.WriteLine(Test(""s""));
+    }
+
+    static string Test(ReadOnlySpan<byte> a) => ""ReadOnlySpan"";
+    static string Test(byte[] a) => ""array"";
+}
+";
+            var comp = CreateCompilation(source, targetFramework: TargetFramework.NetCoreApp, options: TestOptions.DebugExe);
+            CompileAndVerify(comp, expectedOutput: @"array").VerifyDiagnostics();
+        }
+
+        [ConditionalFact(typeof(CoreClrOnly))]
+        public void OverloadResolution_07()
+        {
+            var source = @"
+using System;
+class C
+{
+    static void Main()
+    {
+        System.Console.WriteLine(Test(""s""));
+    }
+
+    static string Test(Span<byte> a) => ""Span"";
+    static string Test(byte[] a) => ""array"";
+}
+";
+            var comp = CreateCompilation(source, targetFramework: TargetFramework.NetCoreApp, options: TestOptions.DebugExe);
+            CompileAndVerify(comp, expectedOutput: @"array").VerifyDiagnostics();
+        }
+
+        [ConditionalFact(typeof(CoreClrOnly))]
+        public void OverloadResolution_08()
+        {
+            var source = @"
+using System;
+class C
+{
+    static void Main()
+    {
+        System.Console.WriteLine(Test(""s""));
+    }
+
+    static string Test(ReadOnlySpan<byte> a) => ""ReadOnlySpan"";
+    static string Test(Span<byte> a) => ""Span"";
+}
+";
+            var comp = CreateCompilation(source, targetFramework: TargetFramework.NetCoreApp, options: TestOptions.DebugExe);
+            CompileAndVerify(comp, expectedOutput: @"Span").VerifyDiagnostics();
+        }
+
+        [ConditionalFact(typeof(CoreClrOnly))]
+        public void OverloadResolution_09()
+        {
+            var source = @"
+using System;
+class C
+{
+    static void Main()
+    {
+        System.Console.WriteLine(Test(""s""));
+    }
+
+    static string Test(ReadOnlySpan<byte> a) => ""ReadOnlySpan"";
+    static string Test(byte[] a) => ""array"";
+    static string Test(Span<byte> a) => ""Span"";
+}
+";
+            var comp = CreateCompilation(source, targetFramework: TargetFramework.NetCoreApp, options: TestOptions.DebugExe);
+            CompileAndVerify(comp, expectedOutput: @"array").VerifyDiagnostics();
+        }
+
+        [ConditionalFact(typeof(CoreClrOnly))]
+        public void OverloadResolution_10()
+        {
+            var source = @"
+using System;
+
+class Program
+{
+    static void Main()
+    {
+        var p = new Program();
+        Console.WriteLine(p.M(""""));
+    }
+
+    public string M(byte[] b) => ""byte[]"";
+}
+
+static class E
+{
+    public static string M(this object o, string s) => ""string"";
+}
+";
+            var comp = CreateCompilation(source, targetFramework: TargetFramework.NetCoreApp, options: TestOptions.DebugExe);
+            CompileAndVerify(comp, expectedOutput: @"byte[]"). // The behavior has changed
+                VerifyDiagnostics();
```

Comment:
Please consider adding:
```csharp
comp = CreateCompilation(source, targetFramework: TargetFramework.NetCoreApp, parseOptions: TestOptions.Regular10, options: TestOptions.DebugExe);
CompileAndVerify(comp, expectedOutput: "string");
```

Reply:
Please add an entry in "docs/compilers/CSharp/Compiler Breaking Changes - DotNet 7.md".

Reply:
The scenario will become an error. We will not change semantics based on language version.

Reply:
> Please add an entry in "docs/compilers/CSharp/Compiler Breaking Changes - DotNet 7.md".

I'll add a PROTOTYPE comment about that for now. We might still change the design around this